hq/test/schedule/unrecognised/IamUnrecognisedUsersTest.scala

```diff
@@ -0,0 +1,125 @@
+package schedule.unrecognised
+
+import com.gu.janus
+import com.gu.janus.model.{ACL, JanusData, SupportACL}
+import model._
+import org.joda.time.{DateTime, Seconds}
+import org.scalatest.{FreeSpec, Matchers}
+import schedule.unrecognised.IamUnrecognisedUsers.{USERNAME_TAG_KEY, getCredsReportDisplayForAccount, getJanusUsernames, unrecognisedUsersForAllowedAccounts}
+import utils.attempt.{FailedAttempt, Failure}
+
+class IamUnrecognisedUsersTest extends FreeSpec with Matchers {
+  val humanUser1 = HumanUser("ade.bimbola", true, AccessKey(NoKey, None), AccessKey(NoKey, None), Green, None, None, List(Tag(USERNAME_TAG_KEY, "ade.bimbola")))
+  val humanUser2 = humanUser1.copy(username = "john.akindele", tags = List(Tag(USERNAME_TAG_KEY, "john.akindele")))
+  val humanUser3 = humanUser1.copy(username = "khadija.omodara", tags = List(Tag(USERNAME_TAG_KEY, "khadija.omodara")))
+  val humanUser4 = humanUser1.copy(username = "nneka.obi", tags = List(Tag(USERNAME_TAG_KEY, "nneka.obi")))
+  val humanUser5 = humanUser1.copy(username = "john.doe", tags = List())
+  val humanUser6 = humanUser1.copy(username = "jane.doe", tags = List(Tag(USERNAME_TAG_KEY, "jane.doe")))
+  val allJanusUsernames = List("ade.bimbola", "john.akindele", "khadija.omodara", "nneka.obi", "jane.doe")
+
+  "findUnrecognisedIamUsers" - {
+    "get janus usernames" in {
+      val dummyJanusData = JanusData(
+        Set(janus.model.AwsAccount("Deploy Tools", "deployTools")),
+        ACL(Map("firstName.secondName" -> Set.empty)),
+        ACL(Map.empty),
+        SupportACL(Map.empty, Set.empty, Seconds.ZERO),
+        None
+      )
+
+      getJanusUsernames(dummyJanusData) shouldEqual List("firstName.secondName")
+    }
+  }
+
+  "getCredsReportDisplayForAccount" - {
+    val failedAttempt: FailedAttempt = FailedAttempt(Failure("error", "error", 500))
+
+    "if an either is a left, aws account does not appear in response" in {
+      val accountCredsLeft = Map(1 -> Left(failedAttempt))
+      getCredsReportDisplayForAccount(accountCredsLeft) shouldEqual Nil
+    }
+    "if an either is a right, aws account included in response" in {
+      val accountCredsRight = Map(1 -> Right(1))
+      getCredsReportDisplayForAccount(accountCredsRight) shouldEqual List((1,1))
+    }
+    "given an empty map, return an empty list" in {
+      getCredsReportDisplayForAccount(Map.empty) shouldEqual Nil
+    }
+    "if all eithers are a left, return an empty list" in {
+      val accountCredsAllLeft = Map(1 -> Left(failedAttempt), 2 -> Left(failedAttempt), 3 -> Left(failedAttempt))
+      getCredsReportDisplayForAccount(accountCredsAllLeft) shouldEqual Nil
+    }
+    "if every either is a right, output list contains same number of elements" in {
+      val accountCredsAllRight = Map(1 -> Right(1), 2 -> Right(2), 3 -> Right(3))
+      getCredsReportDisplayForAccount(accountCredsAllRight) should have length 3
+    }
+  }
+
+  "unrecognisedUsersForAllowedAccounts" - {
```

Comment:
Love these! There are a couple of minor points, but they're a bit tangled with each other so easier to discuss on a call / in person.